cartography/intel/azure/sql.py

```diff
@@ -277,267 +291,237 @@ def load_server_details(neo4j_session, credentials, subscription_id, details, up
-        neo4j_session.run(
-            ingest_restorable_dropped_databases,
-            DatabaseId=data['id'],
-            Name=data['name'],
-            Location=data['location'],
-            DatabaseName=data['database_name'],
-            CreationDate=data['creation_date'],
-            DeletionDate=get_optional_value(data, ['deletion_date']),
-            RestoreDate=data['earliest_restore_date'],
-            Edition=data['edition'],
-            ServiceLevelObjective=data['service_level_objective'],
-            MaxSizeBytes=data['max_size_bytes'],
-            ServerId=data['server_id'],
-            azure_update_tag=update_tag,
-        )
+    neo4j_session.run(
+        ingest_restorable_dropped_databases,
+        restorable_dropped_databases_list=restorable_dropped_databases,
+        azure_update_tag=update_tag,
+    )
 
 
 @timeit
-def _load_failover_groups(neo4j_session, failover_groups, update_tag):
+def _load_failover_groups(
+        neo4j_session: neo4j.Session, failover_groups: List[Dict], update_tag: int
+) -> None:
     """
     Ingest the failover groups details into neo4j.
     """
     ingest_failover_groups = """
-    MERGE (f:AzureFailoverGroup{id: {GroupId}})
+    UNWIND {failover_groups_list} as fg
+    MERGE (f:AzureFailoverGroup{id: fg.id})
     ON CREATE SET f.firstseen = timestamp(), f.lastupdated = {azure_update_tag}
-    SET f.name = {Name},
-    f.location = {Location},
-    f.replicationrole = {Role},
-    f.replicationstate = {State}
-    WITH f
-    MATCH (s:AzureServer{id: {ServerId}})
+    SET f.name = fg.name,
+    f.location = fg.location,
+    f.replicationrole = fg.replication_role,
+    f.replicationstate = fg.replication_state
+    WITH f, fg
+    MATCH (s:AzureSQLServer{id: fg.server_id})
     MERGE (s)-[r:RESOURCE]->(f)
     ON CREATE SET r.firstseen = timestamp()
     SET r.lastupdated = {azure_update_tag}
     """
 
-    for group in failover_groups:
-        neo4j_session.run(
-            ingest_failover_groups,
-            GroupId=group['id'],
-            Name=group['name'],
-            Location=group['location'],
-            Role=group['replication_role'],
-            State=group['replication_state'],
-            ServerId=group['server_id'],
-            azure_update_tag=update_tag,
-        )
+    neo4j_session.run(
+        ingest_failover_groups,
+        failover_groups_list=failover_groups,
+        azure_update_tag=update_tag,
+    )
 
 
 @timeit
-def _load_elastic_pools(neo4j_session, elastic_pools, update_tag):
+def _load_elastic_pools(
+        neo4j_session: neo4j.Session, elastic_pools: List[Dict], update_tag: int
+) -> None:
     """
     Ingest the elastic pool details into neo4j.
     """
     ingest_elastic_pools = """
-    MERGE (e:AzureElasticPool{id: {PoolId}})
+    UNWIND {elastic_pools_list} as ep
+    MERGE (e:AzureElasticPool{id: ep.id})
     ON CREATE SET e.firstseen = timestamp(), e.lastupdated = {azure_update_tag}
```

Comment:
lastupdated should not be in the create set. it should be in update set.

Reply:
done